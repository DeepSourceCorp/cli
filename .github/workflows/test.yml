name: Unit Tests

on:
  push:
    branches: [develop]
  pull_request:
    branches: [develop]

jobs:
  test:
    runs-on: ubuntu-latest
    env:
      NODE_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
      ZEAL_SECRET: ${{secrets.GITHUB_TOKEN}}

    steps:
      - name: Checkout Bifrost
        uses: actions/checkout@v2

      - name: Setting Up Node
        uses: actions/setup-node@v2
        with:
          node-version: '14.x'
          always-auth: true
          registry-url: https://npm.pkg.github.com/
          scope: '@deepsourcelabs'
          cache: 'yarn'

      - name: Install packages
        run: yarn

      - name: Run Test
        run: yarn test --coverage=true --coverageReporters=cobertura

      - name: Install DeepSource CLI
        run: curl https://deepsource.io/cli | sh

      - name: Set DEEPSOURCE_DSN
        run: |
          ./bin/deepsource report --analyzer test-coverage --key javascript --value-file ./coverage/cobertura-coverage.xml
        env:
          DEEPSOURCE_DSN: ${{ secrets.BIFROST_DSN }}
