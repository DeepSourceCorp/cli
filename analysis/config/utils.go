package config

import (
	"log"
	"os"
	"path"
	"path/filepath"
	"strings"

	"github.com/bmatcuk/doublestar/v4"
)

/* Walks the `CODE_PATH` directory and returns all the files other than the ones present
 * in the .git folder for analysis in the form of a string array */
func readAllFiles(codePath string) ([]string, error) {
	fileCount := 0

	allFiles := make([]string, 0)

	err := filepath.Walk(codePath,
		func(path string, fileInfo os.FileInfo, err error) error {
			if err != nil {
				return err
			}
			fileCount++

			/* Check the following before appending to the list of files:
			 * - Should not be a directory
			 * - The walked file should not be present in .git folder
			 * TODO: Check why fileCount != 1 condition is used here */
			if !fileInfo.IsDir() || !strings.HasPrefix(path, filepath.Join(codePath, ".git")) && fileCount != 1 {
				allFiles = append(allFiles, path)
			}
			return nil
		})
	if err != nil {
		return allFiles, err
	}
	return allFiles, nil
}

// Returns the slice of files matching certain glob patterns
func (r *AnalysisRun) getMatchingFiles(patterns []string) ([]string, error) {
	matchedFiles := make([]string, 0)

	// Return all the files if no exclude_patterns are configured
	if len(patterns) == 0 {
		return matchedFiles, nil
	}

	for _, file := range r.AnalysisFiles {
		for i := range patterns {
			match, err := doublestar.Match(path.Join(r.CodePath, patterns[i]), file)
			if err != nil {
				log.Printf("Failed to match the file %s with pattern %s\n", file, patterns[i])
			}
			if match {
				matchedFiles = append(matchedFiles, file)
				break
			}
		}
	}
	return matchedFiles, nil
}

// Filters the analysis files and removes the files matching the exclude_patterns from them
// TODO: Improve the logic here
func (r *AnalysisRun) filterAnalysisFiles() {
	filteredFiles := []string{}
	for _, file := range r.AnalysisFiles {
		for _, excludedFile := range r.ExcludedFiles {
			if file == excludedFile {
				break
			}
			filteredFiles = append(filteredFiles, file)
		}
	}
	r.AnalysisFiles = filteredFiles
}
