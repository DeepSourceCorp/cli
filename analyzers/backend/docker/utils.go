package docker

import (
	"bufio"
	"encoding/json"
	"errors"
	"fmt"
	"io"
	"strings"
)

type ErrorLine struct {
	Error       string      `json:"error"`
	ErrorDetail ErrorDetail `json:"errorDetail"`
}

type ErrorDetail struct {
	Message string `json:"message"`
}

type DockerBuildResponse struct {
	Stream string `json:"stream"`
}

func CheckBuildResponse(rd io.Reader, showAllLogs bool) error {
	var lastLine []byte
	count := 0
	var currentStream string

	scanner := bufio.NewScanner(rd)
	for scanner.Scan() {
		lastLine = scanner.Bytes()
		d := &DockerBuildResponse{}
		err := json.Unmarshal(lastLine, d)
		if err != nil {
			return err
		}
		if d.Stream == "" || d.Stream == "\n" || strings.Contains(d.Stream, "--->") || strings.TrimSuffix(d.Stream, "\n") == currentStream {
			continue
		}
		currentStream = strings.TrimSuffix(d.Stream, "\n")
		if showAllLogs {
			fmt.Println(currentStream)
		}
		count++
	}

	errLine := &ErrorLine{}
	json.Unmarshal([]byte(lastLine), errLine)
	if errLine.Error != "" {
		return errors.New(errLine.Error)
	}
	return scanner.Err()
}


