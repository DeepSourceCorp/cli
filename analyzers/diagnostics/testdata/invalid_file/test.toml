name = "2do-Checker"
