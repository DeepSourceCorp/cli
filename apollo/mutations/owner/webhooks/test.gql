mutation testWebhook($webhookId: ID!) {
  testWebhook(input: { webhookId: $webhookId }) {
    ok
  }
}
