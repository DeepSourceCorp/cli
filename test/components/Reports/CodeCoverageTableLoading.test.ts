import { render } from '@testing-library/vue'

import { CodeCoverageTableLoading } from '~/components/Reports'
import { cartesian, generateBooleanProps } from '~/test/utils'

describe('[[ CodeCoverageTableLoading ]]', () => {
  test('renders code coverage table in loading state with all prop options', () => {
    const isWidgetOptions = generateBooleanProps('isWidget', false)

    cartesian(isWidgetOptions).forEach((propCombination) => {
      const { html } = render(CodeCoverageTableLoading, {
        propsData: { ...propCombination },
        stubs: {
          ZIcon: true,
          ZButton: true
        }
      })

      expect(html()).toMatchSnapshot(JSON.stringify(propCombination))
    })
  })
})
