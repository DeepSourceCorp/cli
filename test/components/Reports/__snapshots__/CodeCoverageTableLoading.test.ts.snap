// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[[ CodeCoverageTableLoading ]] renders code coverage table in loading state with all prop options: {"isWidget":false} 1`] = `
"<section class=\\"overflow-x-auto leading-normal rounded-md border border-ink-200\\">
  <table class=\\"table-fixed cursor w-full\\">
    <thead class=\\"border-b border-ink-200\\">
      <tr class=\\"uppercase text-vanilla-400 text-xs font-semibold\\">
        <th class=\\"py-2.5 text-left w-48 pl-4 sm:w-auto\\">
          Repository
        </th>
        <th class=\\"text-right w-40 pr-2\\">
          <zbutton-stub color=\\"vanilla-400\\" iconcolor=\\"\\" label=\\"\\" loadinglabel=\\"\\" as=\\"button\\" justify=\\"center\\" buttontype=\\"ghost\\" type=\\"button\\" size=\\"x-small\\" disabled=\\"true\\" to=\\"\\">
            <div class=\\"flex items-center gap-x-2 font-semibold\\"><span class=\\"uppercase\\">Line Coverage</span>
              <zicon-stub icon=\\"arrow-up-down\\" size=\\"small\\" position=\\"\\" color=\\"vanilla-400\\"></zicon-stub>
            </div>
          </zbutton-stub>
        </th>
        <th class=\\"text-right w-48 pr-2\\">
          <zbutton-stub color=\\"vanilla-400\\" iconcolor=\\"\\" label=\\"\\" loadinglabel=\\"\\" as=\\"button\\" justify=\\"center\\" buttontype=\\"ghost\\" type=\\"button\\" size=\\"x-small\\" disabled=\\"true\\" to=\\"\\">
            <div class=\\"flex items-center gap-x-2 font-semibold\\"><span class=\\"uppercase\\">Branch Coverage</span>
              <zicon-stub icon=\\"arrow-up-down\\" size=\\"small\\" position=\\"\\" color=\\"vanilla-400\\"></zicon-stub>
            </div>
          </zbutton-stub>
        </th>
      </tr>
    </thead>
    <tbody class=\\"divide-y divide-ink-200\\">
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-4\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
    </tbody>
  </table>
</section>"
`;

exports[`[[ CodeCoverageTableLoading ]] renders code coverage table in loading state with all prop options: {"isWidget":true} 1`] = `
"<section class=\\"overflow-x-auto leading-normal max-h-92 overflow-y-auto\\">
  <table class=\\"table-fixed cursor w-full\\">
    <thead class=\\"border-b border-ink-200\\">
      <tr class=\\"uppercase text-vanilla-400 text-xs font-semibold\\">
        <th class=\\"py-2.5 text-left w-48 pl-5\\">
          Repository
        </th>
        <th class=\\"text-right w-28\\">
          <zbutton-stub color=\\"vanilla-400\\" iconcolor=\\"\\" label=\\"\\" loadinglabel=\\"\\" as=\\"button\\" justify=\\"center\\" buttontype=\\"ghost\\" type=\\"button\\" size=\\"x-small\\" disabled=\\"true\\" to=\\"\\">
            <div class=\\"flex items-center gap-x-2 font-semibold\\"><span class=\\"uppercase\\">Line</span>
              <zicon-stub icon=\\"arrow-up-down\\" size=\\"small\\" position=\\"\\" color=\\"vanilla-400\\"></zicon-stub>
            </div>
          </zbutton-stub>
        </th>
        <th class=\\"text-right w-28 pr-5\\">
          <zbutton-stub color=\\"vanilla-400\\" iconcolor=\\"\\" label=\\"\\" loadinglabel=\\"\\" as=\\"button\\" justify=\\"center\\" buttontype=\\"ghost\\" type=\\"button\\" size=\\"x-small\\" disabled=\\"true\\" to=\\"\\">
            <div class=\\"flex items-center gap-x-2 font-semibold\\"><span class=\\"uppercase\\">Branch</span>
              <zicon-stub icon=\\"arrow-up-down\\" size=\\"small\\" position=\\"\\" color=\\"vanilla-400\\"></zicon-stub>
            </div>
          </zbutton-stub>
        </th>
      </tr>
    </thead>
    <tbody class=\\"divide-y divide-ink-200\\">
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
      <tr class=\\"h-11\\">
        <td class=\\"py-2.5 text-left pl-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px w-48\\"></div>
        </td>
        <td class=\\"py-2.5 pr-4 pl-16 text-right\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
        <td class=\\"py-2.5 pl-24 text-right pr-5\\">
          <div class=\\"bg-ink-300 animate-pulse rounded-md h-5 mb-px\\"></div>
        </td>
      </tr>
    </tbody>
  </table>
</section>"
`;
